### PR TITLE
[rom_ext] Remove Signature Identifier

### PR DESCRIPTION
**Pull request by lenary**
This was left over from https://github.com/lowRISC/opentitan/pull/3733#issuecomment-726928866.

There are three other changes:
- One adds verbose logging to the generator script, which makes it easier to understand how layout is being performed when the manifest is changed.
- One updates the `manifest.md` to use the format from #4305.
- One updates the assembly which is used for the skeleton of the ROM_EXT in the elf images (which we plan to auto-generate at some point).

### CODE REVIEW

File: util/rom-ext-manifest-generator.py [hunk 1]
```diff
@@ -92,23 +94,47 @@ def generate_defines(fields):
...
-        current_offset_bytes = (current_offset_bytes + required_alignment_bytes - 1) \
+        new_offset_bytes = (current_offset_bytes + required_alignment_bytes - 1) \
             & ~(required_alignment_bytes - 1)
 
+        if new_offset_bytes != current_offset_bytes and verbose:
+            print("0x{:04x} - 0x{:04x}: - (realignment) -".format(
+                current_offset_bytes, new_offset_bytes))
+
+        current_offset_bytes = new_offset_bytes
+        current_offset_alignment = current_offset_bytes & -current_offset_bytes
```
Comment (silvestrst) [inline util/rom-ext-manifest-generator.py:107]: Could you maybe in a comment drop an example with `4` and `-4` (...00000100 & ...11111100). It is pretty obvious what it does, but I didn't come across this pattern match.

File: sw/device/rom_exts/manifest.md [hunk 2]
```diff
@@ -6,74 +6,74 @@ Based on the requirements outlined in [boot.md](./boot)
...
 |                                                               |
-+                        Image Timestamp                        +
++                        image_timestamp                        +
 |                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                 Signature Algorithm Identifier                |
+|                 signature_algorithm_identifier                |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                       Signature Exponent                      |
+|                            exponent                           |
```
Comment (silvestrst) [inline sw/device/rom_exts/manifest.md]: Requires naming changes as per #4361 .

File: sw/device/rom_exts/manifest.md [hunk 3]
```diff
@@ -6,74 +6,74 @@ Based on the requirements outlined in [boot.md](./boot)
...
++                    peripheral_lockdown_info                   +
 |                                                               |
 +                                                               +
 |                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |                                                               |
 +                                                               +
 |                                                               |
-+                Signature Public Key (3072 bits)               +
++                signature_public_key (3072 bits)               +
```
Comment (silvestrst) [inline sw/device/rom_exts/manifest.md:52]: Requires naming changes as per #4361 .

File: sw/device/rom_exts/manifest.md (hunk 2 above)
Comment (silvestrst) [reply sw/device/rom_exts/manifest.md]: Ooops, this file is auto generated. Is it possible to add this information to the top of the file (auto generated but also manually committed)?

File: sw/device/rom_exts/manifest.md (hunk 3 above)
Comment (silvestrst) [reply sw/device/rom_exts/manifest.md:52]: Ooops, this file is auto generated. Is it possible to add this information to the top of the file (auto generated but also manually committed)?

File: sw/device/rom_exts/manifest.hjson [hunk 4]
```diff
@@ -96,7 +91,7 @@
     },
     {
       name: "signature_public_key",
-      desc: '''ROM_EXT Manifest sifnature public key.
+      desc: '''ROM_EXT Manifest sifnature public key modulus.
```
Comment (silvestrst) [inline sw/device/rom_exts/manifest.hjson:94]: I changed the field name as well in #4361 .

**Review by silvestrst — COMMENTED** [review]
@lenary thanks! Not a big fan of debug code inside the production code, but in this case I think it's fine ("verbose" change).

Is it possible to put debug blocks like:
```
                if verbose:
                    print("0x{:04x} - 0x{:04x}: {} (alignment: {} reqd: {})".
                          format(current_offset_bytes,
                                 current_offset_bytes + size_bytes,
                                 field['name'], current_offset_alignment,
```
In their own function like `debug_printout`. I can see that these debug blocks are slightly different form each other, so If it is hard to combine them - I think it is fine to leave it as it is.

I am also not entirely sure how this change related to #4361 and which supposed to be landed first.

File: util/rom-ext-manifest-generator.py (hunk 1 above)
Comment (lenary) [reply util/rom-ext-manifest-generator.py]: A better example is `6`, which should have output `2`.

**Comment by lenary** [comment]
Rebased. Tagged @felixmiller to check I didn't stomp all over his rename. This was a nightmare to rebase.

File: util/rom-ext-manifest-generator.py (hunk 1 above)
Comment (silvestrst) [reply util/rom-ext-manifest-generator.py]: Sounds good.

File: sw/device/rom_exts/rom_ext_manifest.S [hunk 5]
```diff
@@ -38,22 +39,20 @@ image_version:
...
-// Reserved
-  .word 0x0
+// Overriden by signing script?
+image_usage_constraints:
+  .rept 32
+  .byte 0x0
+  .endr
 
-// Maybe: Overriden by signing script?
+// Overriden by signing script?
```
Comment (silvestrst) [inline sw/device/rom_exts/rom_ext_manifest.S:55]: Is the question here - whether this information should be populated at the time of image creation or at the time of signing? The same goes for `image_usage_constraints`.

**Review by silvestrst — APPROVED** [review]
@lenary thanks!

Looks good to me!

File: sw/device/rom_exts/rom_ext_manifest.S (hunk 5 above)
Comment (lenary) [reply sw/device/rom_exts/rom_ext_manifest.S:55]: Sorry, the question mark should go. This should be overridden by the signing script. The same goes for `usage_constraints.

File: sw/device/rom_exts/rom_ext_manifest.S [hunk 6]
```diff
@@ -38,22 +39,20 @@ image_version:
 image_timestamp:
   .dword 0x0
 
-// Overridden by signing script
-image_signature_algorithm:
-  .word 0x0
-
 // Overriden by signing script
-image_signature_exponent:
+image_signature_public_key_exponent:
```
Comment (felixmiller) [inline sw/device/rom_exts/rom_ext_manifest.S:43]: This should rather read `image_signature_key_public_exponent`. I think what is meant is the _public exponent_ of the _signature key_ of the _image_.

Comment (lenary) [reply sw/device/rom_exts/rom_ext_manifest.S:43]: Yeah, sorry. 

The plan in the long-run is to auto-generate this file from manifest.hjson, which should make your life easier for this kind of change :)

File: util/rom-ext-manifest-generator.py [hunk 7]
```diff
@@ -170,14 +204,14 @@ def generate_format_description(regions, output_dir):
 
     output_path = output_dir / 'manifest.txt'
 
-    truncate_length = 16 # bytes
+    truncate_length = 16  # bytes
     bits_in_byte = 8
 
     verbose_regions = []
```
Comment (alphan) [inline util/rom-ext-manifest-generator.py:210]: nit: This is not related to the new `verbose` flag, right? I know this was already named `verbose_regions` but IMHO using the `protocol_input` from the beginning or something like `protocol_regions` would be clearer.

**Review by alphan — COMMENTED** [review]
Thanks @lenary! Just two nits. Also, how about using the `logging` library instead of passing using the `verbose` flag to `generate_defines`?

File: util/rom-ext-manifest-generator.py (hunk 7 above)
Comment (lenary) [reply util/rom-ext-manifest-generator.py:210]: Yeah, this is probably a bad name. I'll do a small PR this afternoon to fix this.

**Comment by lenary** [comment]
@alphan I avoided the logging library for no good reason. I'll address that in today's PR too.
